File: 397-integer-replacement/integer-replacement.py
class Solution(object):
    def integerReplacement(self, n):
        """
        :type n: int
        :rtype: int
        """
        c=0
        while n!=1:
            if (n&1)==0:
                n//=2
            else:
                if n==3 or n%4==1: n-=1
                else: n+=1
            c+=1
        return c
        

File: 397-integer-replacement/integer-replacement.java
class Solution {
    public int integerReplacement(int n) {
        int c=0;
        long l=n;
        while(l!=1){
            if((l&1)==0) l/=2;
            else{
                if(l==3 || l%4==1) l--;
                else l++;
            }
            c++;
        }
        return c;
    }
}